felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -897,6 +964,14 @@ func (m *bpfEndpointManager) applyPolicy(ifaceName string) error {
 
 	applyTime := time.Since(startTime)
 	log.WithField("timeTaken", applyTime).Info("Finished applying BPF programs for workload")
+
+	m.ifacesLock.Lock()
+	m.withIface(ifaceName, func(iface *bpfInterface) (forceDirty bool) {
+		iface.dpState.isReady = true
```

Comment:
Think this needs to be set back to false somewhere.  It's also critical that `iface` is completely zeroed out once the wep and the interface are both gone.  Otherwise we leak.  The `withIface` method checks if the `iface` is the zero struct and cleans it up if so.